### PR TITLE
Update PO success modal

### LINKED ISSUE #7579: PO success modal needs a new iteration
> The modal feels too salesy for the user who’s already invested. Why do they need to be sold on this step? For something that’s meant to be quick and easy, there’s too much reading required. Simplify the copy to be more straightforward.

More details paJDYF-bax-p2

### PR DESCRIPTION
**Pull request by ismaeldcom**
Fixes #7579

#### Changes proposed in this Pull Request
- Implement new PO success modal design (Figma: VLWDmeuIXQ7XAK4flyqTFO-fi-182_17764)
- Install the `canvas-confetti` package and type definitions.
- Create `ConfettiAnimaiton` component. We were waiting for the new release of the `@automatic/components` package to use the existing component there. However, after testing, I found that there are some typing issues, our bundle will grow more than expected, and the component doesn't fit well with the proposed design. So it's better to use our own component, since it's quite small and simple.

#### Testing instructions
Confirm that the implemented design matches the one in Figma, ensuring it's properly formatted in mobile size.

As this doesn't affect the logic, the best way to test it is to set the following condition to true:
https://github.com/Automattic/woocommerce-payments/blob/5634b1789b4e74baa489d4c27f3779fe03d18183/client/overview/index.js#L213
And comment this line:
https://github.com/Automattic/woocommerce-payments/blob/5634b1789b4e74baa489d4c27f3779fe03d18183/client/overview/modal/progressive-onboarding-eligibility/index.tsx#L65

#### Screenshots
| Before | After | After Mobile |
|--------|--------|--------|
|Screenshot 2024-01-10 at 12 36 54|Screenshot 2024-01-19 at 12 06 06|Screenshot 2024-01-19 at 12 06 41|

**Confetti animation**

https://github.com/Automattic/woocommerce-payments/assets/7670276/520c671b-0b02-438f-8637-a9712fc88131

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8005` or branch name `update/7579-po-success-modal` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/7579-po-success-modal) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f35f4b7b09808651b59ad583de1302a7d939c4d2
- Build time: 2024-01-19 15:55:22 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by ismaeldcom** [comment]
Hey @elizaan36! When you have time, I'd appreciate a quick design review, to confirm that the design matches yours, especially around the confetti animation. I've tried to set it up to match as much as possible the static reference in Figma 🙂

**Review by vladolaru — APPROVED** [review]
Oh wow! Such nice confetti. I refreshed a couple of times just to experience it more :D The code changes look good and the testing instructions checked out.

Great job, Ismael!

**Review by oaratovskyi — APPROVED** [review]
Approve! Nice-looking popup and animation :shipit:

File: client/components/confetti-animation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import confetti from 'canvas-confetti';
+
+const defaultColors = [ '#889BF2', '#C3CDF9', '#6079ED' ];
+
+const fireConfetti = ( colors: string[] ) => {
```
Comment (elizaan36) [inline client/components/confetti-animation/index.tsx:9]: +1 looks really great! I realize there wasn't a design animation provided so if it's simple to do could we start the confetti with a wider spread and a little higher up on the screen? I'm thinking that the current starting point is too tight and the "explosion" should be a bit more subtle.

Comment (elizaan36) [reply client/components/confetti-animation/index.tsx:9]: In the designs I was envisioning the animation would start at what ~1sec in the video looks like. 
image

**Comment by elizaan36** [comment]
Apart from the minor comment about the confetti starting point the modal implementation looks good to me! Thank you for sharing the mobile layout.

**Comment by vladolaru** [comment]
@ismaeldcom can you create a follow up issue and PR with minor adjustments @elizaan36 mentioned? 

In the interest of getting this shipped I went ahead and merged this PR in its current state.

**Comment by elizaan36** [comment]
Hi @bgrgicak 👋 Poli mentioned that you may be working on a reusable component for the confetti so I'm including you in this thread to help connect the dots a bit. :) Let us know if you have any more info.

**Comment by bgrgicak** [comment]
@elizaan36 I will have a PR tomorrow with the confetti component. `@automatic/components` is now updated so WooCommerce can use it.

**Comment by bgrgicak** [comment]
Here is the [confetti component PR](https://github.com/woocommerce/woocommerce/actions/runs/7621522662/job/20758008287?pr=43990).

**Pull request merged**